### PR TITLE
[2.3.2.r1.4] Linux 4.9.164

### PR DESCRIPTION
**Pull request by nathanchance**

### CODE REVIEW

File: include/linux/cpufreq.h [hunk 1]
```diff
@@ -235,20 +235,12 @@ __ATTR(_name, _perm, show_##_name, NULL)
 static struct freq_attr _name =			\
 __ATTR(_name, 0644, show_##_name, store_##_name)
 
-struct global_attr {
```
Comment (MarijnS95) [inline include/linux/cpufreq.h:238]: The removal of this struct needs a slight modification to `cpufreq_interactive.c` in order to compile properly:

```diff
diff --git a/drivers/cpufreq/cpufreq_interactive.c b/drivers/cpufreq/cpufreq_interactive.c
index 8d6abdea0ae9..059282b0bb48 100644
--- a/drivers/cpufreq/cpufreq_interactive.c
+++ b/drivers/cpufreq/cpufreq_interactive.c
@@ -1388,7 +1388,7 @@ static ssize_t store_use_migration_notif(
  */
 #define show_gov_pol_sys(file_name)                                    \
 static ssize_t show_##file_name##_gov_sys                              \
-(struct kobject *kobj, struct attribute *attr, char *buf)              \
+(struct kobject *kobj, struct kobj_attribute *attr, char *buf)         \
 {                                                                      \
        return show_##file_name(common_tunables, buf);                  \
 }                                                                      \
@@ -1401,7 +1401,7 @@ static ssize_t show_##file_name##_gov_pol                         \
 
 #define store_gov_pol_sys(file_name)                                   \
 static ssize_t store_##file_name##_gov_sys                             \
-(struct kobject *kobj, struct attribute *attr, const char *buf,                \
+(struct kobject *kobj, struct kobj_attribute *attr, const char *buf,   \
        size_t count)                                                   \
 {                                                                      \
        return store_##file_name(common_tunables, buf, count);          \
@@ -1437,7 +1437,7 @@ show_store_gov_pol_sys(fast_ramp_down);
 show_store_gov_pol_sys(enable_prediction);
 
 #define gov_sys_attr_rw(_name)                                         \
-static struct global_attr _name##_gov_sys =                            \
+static struct kobj_attribute _name##_gov_sys =                         \
 __ATTR(_name, 0644, show_##_name##_gov_sys, store_##_name##_gov_sys)
 
 #define gov_pol_attr_rw(_name)                                         \
@@ -1466,7 +1466,7 @@ gov_sys_pol_attr_rw(ignore_hispeed_on_notif);
 gov_sys_pol_attr_rw(fast_ramp_down);
 gov_sys_pol_attr_rw(enable_prediction);
 
-static struct global_attr boostpulse_gov_sys =
+static struct kobj_attribute boostpulse_gov_sys =
        __ATTR(boostpulse, 0200, NULL, store_boostpulse_gov_sys);
 
 static struct freq_attr boostpulse_gov_pol =
```

Do you want to amend that, or should I submit it as a separate commit to your branch?

Comment (nathanchance) [reply include/linux/cpufreq.h:238]: Let's do it as a separate commit so it is clear what is going on (that would be needed for forward porting purposes in the future). I'll cherry pick it and push it here when you're done.

Comment (MarijnS95) [reply include/linux/cpufreq.h:238]: Sounds good. Pushed the commit here https://github.com/MarijnS95/kernel/commit/8cb34146b2ff9e2c46882d35eb30b7ff3c05f343 (I hope the description is allright, kernel novice here :sweat_smile: - edit: Reading it again, noticed the word `to` is missing inbetween `changes cpufreq_interactive` :see_no_evil:).
